docs/product-user-guide/src/main/asciidoc/guided-rules-WHEN-proc.adoc

```diff
@@ -1,50 +1,54 @@
-+
-.Multiple Field Constraint on Formula Expression
-image::loanapp.png[]
+. In the guided rule designer, click the plus icon (image:5686.png[]) on the right side of the `WHEN` section.
++
+The *Add a condition to the rule* window with the available condition elements opens.
++
+.Add a condition to the rule
+image::5687.png[Add a condition to the rule]
++
+The list includes the data objects from the *Data Objects* tab of the guided rule designer, any DSL objects defined for the package (if you selected *Use Domain Specific Language (DSL)* when you created this guided rule), and the following standard options:
+
+* *The following does not exist:* Use this to specify facts and constraints that must not exist.
+* *The following exists:* Use this to specify facts and constraints that must exist. This option is triggered on only the first match, not subsequent matches.
+* *Any of the following are true:* Use this to list facts and constraints that must be true.
+* *From:* Use this to define a `From` conditional element for the rule.
+* *From Accumulate:* Use this to define an `Accumulate` conditional element for the rule.
+* *From Collect:* Use this to define a `Collect` conditional element for the rule.
+* *From Entry Point:* Use this to define an `Entry Point` for the pattern.
+* *Free form DRL:* Use this to insert a free-form DRL field where you can define condition elements freely, without the guided rule designer.
++
+*<@Michael (1 of 6): I was going to put a note here about template keys, but they don't apply to free form DRL within the guided rule designer, right? Or do they? Since I don't see any option in the guided editor either for template keys, when I modify values, I assume it's safe to say that template keys just don't apply to guided rules, only to guided decision tables, guide rule templates, and decision tables, where you can define the values to be interpolated with the keys. Maybe DRL rules too, but I'm not there yet. That right?>*
++
+////
+.Template keys in free form DRL
+[NOTE]
+====
+With free form DRL, you have the flexibility to add custom data types, such as template keys. Template keys are placeholder variables in your field constraints that are interpolated with a specified value when the DRL rules are generated. You can add a template key value to facts in the format `@key`. For example, `Person( age > @{age} )` creates an `$age` column in the decision table.
+====
+////
+. Choose a condition element (for example, *LoanApplication*) and click *Ok*.
+. Click the condition element in the guided rule designer and use the *Modify constraints for LoanApplication* window to add a restriction on a field, apply multiple field constraints, add a new formula style expression, apply an expression editor, or set a variable name.
+. Modify the condition to suit your needs. For example, you can set a variable name to help define the condition, or you can add a restriction.
```

Comment:
It might be worth clarifying ".. to help define the condition..." Really it allows the User to identify a Fact/Field in other constructs within the Guided Rule. For example, I could set the variable of a ```Person``` to ```p``` and then reference ```p``` in another constraint on another Fact to check that a ```Person``` is the parent of another: ```p : Person() Person( parent == p )```

Reply:
@manstis gotcha. I'd like to use a LoanApplication example to stick to this scenario. I came up with this, but don't know if it's quite right. This close?

"A variable name enables you to identify a fact or field in other constructs within the guided rule. For example, you could set the variable of `LoanApplication` to `a` and then reference `a` in a separate `Bankruptcy` constraint that specifies which `LoanApplication` the bankruptcy is based on: `a : LoanApplication() Bankruptcy( application == a )`."

Reply:
@sterobin On face value it is correct; however none of the Data Objects in the mortgage example reference other domain-specific Data Objects i.e. ```Bankruptcy``` does not have a field ```application```. I had a quick check and there are none in the project; so IDK whether its better to give a hypothetical example vs one that gives the impression it is possible to create?

I'll leave the decision to you.

Reply:
@manstis , shoot, realized I also have to come up with a Customer example for the rule templates doc. I feel like this one's even more incorrect. Let me know what you think.

"A variable name enables you to identify a fact or field in other constructs within the guided rule. For example, you could set the variable of `Customer` to `c` and then reference `c` in a separate `Applicant` constraint that specifies that the `Customer` is the `Applicant`: `c : Customer() Applicant( person == c )`."

Reply:
@sterobin Looks OK to me.. I'd propose you remove one of the colons to improve readability.

"A variable name enables you to identify a fact or field in other constructs within the guided rule. For example, you could set the variable of ```Customer``` to ```c``` and then reference ```c``` in a separate ```Applicant``` constraint that specifies that the ```Customer``` is the ```Applicant``` **i.e.** ```c : Customer() Applicant( person == c )```."

Reply:
Got it. Thanks Michael.